src/cryptsetup/cryptsetup-generator.c

```diff
@@ -106,11 +152,33 @@ static int create_disk(
                 "DefaultDependencies=no\n"
                 "Conflicts=umount.target\n"
                 "IgnoreOnIsolate=true\n"
-                "After=%s\n",
+                "After=%s\n\n",
                 netdev ? "remote-fs-pre.target" : "cryptsetup-pre.target");
 
+        if (keydev) {
+                _cleanup_free_ char *unit = NULL, *mount = NULL, *p = NULL;
+
+                if (!password)
+                        return log_error_errno(-EINVAL, "Path to password file not specified: %m");
+
+                r = generate_keydev_mount(name, keydev, &unit, &mount);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate keydev mount unit: %m");
+
+                p = prefix_root(mount, password_escaped);
+                if (!p)
+                        return log_oom();
+                free_and_replace(password_escaped, p);
+
+                /* Cleanup the keydev mount after the attach */
+                fprintf(f,
+                        "[Service]\n"
+                        "ExecStartPost=" UMOUNT_PATH " %s\n\n", mount);
```

Comment:
Yeah, this part I don't particularly like. I would love to see this expressed as unit dependencies, but maybe I'm overthinking this. After all, we do explicit mounts from other units. Let's do it like this unless somebody comes up with a better approach.

Reply:
I thought about that, but I figured it is probably not possible because systemd-cryptsetup unit is ```RemainAfterExit=yes``` oneshot and hence it stays around. Thus it always references the mount unit and StopWhenUneeded wouldn't work. Since it stays around also Requires/BindsTo/PartOf wouldn't do the trick.

Reply:
Ack.